### PR TITLE
made new branch with all the changes from main

### PR DESCRIPTION
**Pull request by vsych321**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -69,6 +70,33 @@
             <artifactId>mapstruct</artifactId>
             <version>${version.mapstruct}</version>
         </dependency>
+        <dependency>
+            <groupId>org.jetbrains</groupId>
+            <artifactId>annotations</artifactId>
+            <version>RELEASE</version>
+            <scope>compile</scope>
+        </dependency>
```
Comment (liliia-ponomarenko) [inline pom.xml]: why do you need this dependency?

File: pom.xml [hunk 2]
```diff
@@ -128,4 +156,4 @@
             </plugin>
         </plugins>
     </build>
-</project>
+</project>
```
Comment (liliia-ponomarenko) [inline pom.xml]:
```suggestion
</project>

```

File: src/main/java/bookstore/config/SecurityConfig.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+@EnableMethodSecurity
+@RequiredArgsConstructor
+@Configuration
+public class SecurityConfig {
+    private final JwtAuthenticationFilter jwtAuthenticationFilter;
+
```
Comment (liliia-ponomarenko) [inline src/main/java/bookstore/config/SecurityConfig.java]:
```suggestion
```

File: src/main/java/bookstore/controller/AuthenticationController.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Authentication management", description = "Endpoints for managing authentication")
+@RestController
+@RequiredArgsConstructor
+@Validated
```
Comment (liliia-ponomarenko) [inline src/main/java/bookstore/controller/AuthenticationController.java]:
```suggestion
```
you don’t  use this annotation

File: src/main/java/bookstore/controller/BookController.java [hunk 5]
```diff
@@ -32,29 +32,29 @@
...
-    @ResponseStatus(HttpStatus.ACCEPTED)
+    @ResponseStatus(HttpStatus.OK)
     @GetMapping("/{id}")
     @Operation(summary = "Get book by id", description = "Get available book by certain id")
     public BookDto getBookById(@PathVariable @Positive Long id) {
         return bookService.getById(id);
     }
 
-    @ResponseStatus(HttpStatus.CREATED)
+    @ResponseStatus(HttpStatus.OK)
```
Comment (liliia-ponomarenko) [inline src/main/java/bookstore/controller/BookController.java]: return CREATED

File: src/main/java/bookstore/controller/BookController.java [hunk 6]
```diff
@@ -32,29 +32,29 @@
...
     @PostMapping
     @Operation(summary = "Create a new book", description = "Create a new book")
     public BookDto createBook(@RequestBody @Valid CreateBookRequestDto requestDto) {
         return bookService.save(requestDto);
     }
 
     @PutMapping("/{id}")
-    @ResponseStatus(HttpStatus.ACCEPTED)
+    @ResponseStatus(HttpStatus.OK)
```
Comment (liliia-ponomarenko) [inline src/main/java/bookstore/controller/BookController.java:62]: OK it’s  default status

File: src/main/java/bookstore/dto/user/UserLoginRequestDto.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
+package bookstore.dto.user;
+
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotEmpty;
+import jakarta.validation.constraints.Size;
+
+public record UserLoginRequestDto(
+        @NotBlank
+        @Size(min = 8)
```
Comment (liliia-ponomarenko) [inline src/main/java/bookstore/dto/user/UserLoginRequestDto.java]: should you add max size?

File: src/main/java/bookstore/dto/user/UserLoginRequestDto.java [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+        @Email
+        String email,
+
+        @NotEmpty
+        @Size(min = 8)
```
Comment (liliia-ponomarenko) [inline src/main/java/bookstore/dto/user/UserLoginRequestDto.java]: the same

File: src/main/java/bookstore/dto/user/UserResponseDto.java [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+public record UserResponseDto(
+        Long id,
+        String email,
+
+        String firstName,
+
+        String lastName,
+
+        String shippingAddress
+
```
Comment (liliia-ponomarenko) [inline src/main/java/bookstore/dto/user/UserResponseDto.java]:
```suggestion
```

File: src/main/java/bookstore/dto/user/UserResponseDto.java [hunk 10]
```diff
@@ -0,0 +1,14 @@
+package bookstore.dto.user;
+
+public record UserResponseDto(
+        Long id,
```
Comment (liliia-ponomarenko) [inline src/main/java/bookstore/dto/user/UserResponseDto.java:4]:
```suggestion
        Long id,
        
```
I'm not a fan of empty lines, but if you like it, stick to the same style in all your classes

File: src/main/java/bookstore/entity/User.java [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+    @JoinTable(
+            name = "user_role",
+            joinColumns = @JoinColumn(name = "user_id"),
+            inverseJoinColumns = @JoinColumn(name = "role_id")
+    )
+    @ToString.Exclude
+    @EqualsAndHashCode.Exclude
+    private Set<Role> roles = new HashSet<>();
+
+    @Column(name = "is_deleted")
```
Comment (liliia-ponomarenko) [inline src/main/java/bookstore/entity/User.java]:
```suggestion
    @Column(name = "is_deleted", nullable = false)
```

File: src/main/java/bookstore/security/CustomUserDetailsService.java [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+
+@RequiredArgsConstructor
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        return userRepository.findByEmail(username).orElseThrow(
+                () -> new RuntimeException("Can't find user my email"));
```
Comment (liliia-ponomarenko) [inline src/main/java/bookstore/security/CustomUserDetailsService.java]:
```suggestion
                () -> new UsernameNotFoundException("Can't find user my email"));
```
and remove it from the signature

File: src/main/java/bookstore/security/JwtAuthenticationFilter.java [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+                    userDetails, null, userDetails.getAuthorities()
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader(HttpHeaders.AUTHORIZATION);
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
```
Comment (liliia-ponomarenko) [inline src/main/java/bookstore/security/JwtAuthenticationFilter.java]: better make constant and you can use length instead magic number 7

File: src/main/java/bookstore/service/impl/UserServiceImpl.java [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+        if (userRepository.existsByEmail(request.email())) {
+            throw new RegistrationException("Unable to complete registration");
+        }
+        User user = userMapper.toUser(request);
+        user.setRoles(Collections.singleton(roleRepository
+                .findRoleByName(Role.RoleName.ROLE_USER)));
+        user.setPassword(passwordEncoder.encode(request.password()));
+        user.setEmail(request.email());
+        User savedUser = userRepository.save(user);
+        return userMapper.toUserResponse(savedUser);
```
Comment (liliia-ponomarenko) [inline src/main/java/bookstore/service/impl/UserServiceImpl.java]: can be an inline variable

File: src/main/java/bookstore/validation/FiledMatcherValidator.java [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+public class FiledMatcherValidator implements ConstraintValidator<FieldMatch,
+        UserRegistrationRequestDto> {
+
+    @Override
+    public boolean isValid(UserRegistrationRequestDto requestDto,
+                           ConstraintValidatorContext constraintValidatorContext) {
+        return requestDto.password().equals(requestDto.repeatPassword());
```
Comment (liliia-ponomarenko) [inline src/main/java/bookstore/validation/FiledMatcherValidator.java]: how about not null check?

File: src/main/resources/db/changelog/changes/02-create-users-table.yaml [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+                    nullable: false
+              - column:
+                  name: shipping_address
+                  type: varchar(255)
+              - column:
+                  name: is_deleted
+                  type: boolean
+                  defaultValueBoolean: false
+                  constraints:
+                    nullable: false
```
Comment (liliia-ponomarenko) [inline src/main/resources/db/changelog/changes/02-create-users-table.yaml]: one empty line (check other files)

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let’s improve your solution ;)

File: src/main/java/bookstore/controller/BookController.java (hunk 6 above)
Comment (vsych321) [reply src/main/java/bookstore/controller/BookController.java:62]: That's why I use it

File: src/main/java/bookstore/validation/FiledMatcherValidator.java [hunk 17]
```diff
@@ -12,6 +12,9 @@ public class FiledMatcherValidator implements ConstraintValidator<FieldMatch,
     @Override
     public boolean isValid(UserRegistrationRequestDto requestDto,
                            ConstraintValidatorContext constraintValidatorContext) {
-        return requestDto.password().equals(requestDto.repeatPassword());
+        if (requestDto != null) {
+            return requestDto.password().equals(requestDto.repeatPassword());
+        }
+        return false;
```
Comment (liliia-ponomarenko) [inline src/main/java/bookstore/validation/FiledMatcherValidator.java:18]: can you use the ternary operator?

**Review by liliia-ponomarenko — APPROVED** [review]
Great! 😌

**Pull request merged**